### PR TITLE
script: Ensure we don't ignore reflows for queries that bail out due to the viewport size not yet present

### LINKED ISSUE #11223: Reddit.com doesn't load completely within browser.html
Duplicate of https://github.com/browserhtml/browserhtml/issues/1037 as this is apparently a servo bug.

Often the mozbrowser iframe doesn't get the `loadend` event when loading reddit.com.

It appears to happen only with browserhtml.

This is what I managed to figure out so far:

- In document.rs, in `finish_load`, for the `reddit.com` pipeline, `DocumentLoadsComplete` is properly sent. Which means that `loader.is_blocked()` return `false`.
- then, in `script_thread.rs`, in `handle_loads_complete`, `doc.loader().is_blocked()` returns `true`, preventing finishing the load.

So I guess we need to figure out why sometimes the loader gets blocked.

### PR DESCRIPTION
**Pull request by emilio**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #11223 

- [ ] There are tests for these changes.

This fixes #11223.

Note that this didn't happen in the root pipeline because we explicitly set the
window size in that case.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11545)

### CODE REVIEW

**Comment by emilio** [comment]
r? @pcwalton

Don't know what would be a good way to test it, what infrastructure exists for a minimal test-case like https://github.com/servo/servo/issues/11223#issuecomment-223037836?

cc @jdm @nox @paulrouget

File: components/script/dom/window.rs [hunk 1]
```diff
@@ -1019,26 +1019,33 @@ impl Window {
     /// and no reflow is performed. If reflow is suppressed, no reflow will be
     /// performed for ForDisplay goals.
     ///
-    /// TODO(pcwalton): Only wait for style recalc, since we have off-main-thread layout.
-    pub fn force_reflow(&self, goal: ReflowGoal, query_type: ReflowQueryType, reason: ReflowReason) {
+    /// TODO(pcwalton): Only wait for style recalc, since we have
+    /// off-main-thread layout.
+    ///
+    /// Returns true if the reflow was not suppressed, false otherwise.
```
Comment (pcwalton) [inline components/script/dom/window.rs]: How about "returns true if the layout actually happened or false if it was suppressed"? The double negative is a bit confusing :)

File: components/script/dom/window.rs [hunk 2]
```diff
@@ -1100,18 +1107,32 @@ impl Window {
...
+    /// method will wait for the layout thread to complete (but see the `TODO`
+    /// below). If there is no window size yet, the page is presumed invisible
+    /// and no reflow is performed.
     ///
-    /// TODO(pcwalton): Only wait for style recalc, since we have off-main-thread layout.
-    pub fn reflow(&self, goal: ReflowGoal, query_type: ReflowQueryType, reason: ReflowReason) {
+    /// TODO(pcwalton): Only wait for style recalc, since we have
+    /// off-main-thread layout.
+    ///
+    /// Returns true if the reflow was not supressed, false otherwise. This
```
Comment (pcwalton) [inline components/script/dom/window.rs]: Ditto.

Comment (pcwalton) [inline components/script/dom/window.rs]: Also typo: supressed -> suppressed

**Comment by pcwalton** [comment]
r=me with those nits addressed

**Comment by emilio** [comment]
@bors-servo: r=pcwalton

Thanks for the super-fast review! :)

**Comment by jdm** [comment]
A test showing that it didn't deadlock seems reasonable...

**Pull request merged**